backend/src/nodes/ncnn_nodes.py

```diff
@@ -87,8 +93,8 @@ def upscale(
                 staging_vkallocator=staging_vkallocator,
                 max_depth=tile_mode if tile_mode > 0 else None,
             )
-            # blob_vkallocator.clear() # this slows stuff down
-            # staging_vkallocator.clear() # as does this
+            blob_vkallocator.clear()
+            staging_vkallocator.clear()
```

Comment:
Memory won't be cleared if `ncnn_auto_split_process` throws. Can you use `with` on the allocators?

Reply:
I don't think so, but I'll check. We could make a custom context manager for it

Reply:
Confirmed it does not have one in the pybind. We can either make our own context manager for it, or just check if they are not none in the except and call clear on them there

Reply:
I'd rather have a context manager. There's no need to do `try except` when python has a language feature for exactly this purpose.

Reply:
Well luckily that's what I did :)

Reply:
Oh